### PR TITLE
media/utils: Add check for ID3v2 tag in mp3 file format

### PR DESCRIPTION
**Pull request by aadotverma**
media/utils: Add check for ID3v2 tag in mp3 file format

### CODE REVIEW

**Comment by vibhor-m** [comment]
Code changes are OK.

**Review by Taejun-Kwon — COMMENTED** [review]
I think you can optimize isMP3 for both of streambuffer & file.

I think it will not take long time but if you think it need more time then you can put some TODO comment.

**Comment by vibhor-m** [comment]
> I think you can optimize isMP3 for both of streambuffer & file.
> 
> I think it will not take long time but if you think it need more time then you can put some TODO comment.

changes can be done now

**Review by vibhor-m — APPROVED** [review]
code changes are OK

File: framework/src/media/utils/MediaUtils.cpp [hunk 1]
```diff
@@ -1002,23 +1028,19 @@ bool file_header_parsing(FILE *fp, audio_type_t audioType, unsigned int *channel
...
-			medvdbg("string id3v2 tag found\n");
-			/* todo: check for extended header and parse title, artist etc details */
-			fseek(fp, 128, SEEK_SET);
-			break;
-		} else {
-			medvdbg("string id3v2 tag not found\n");
-			fseek(fp, 0, SEEK_SET);
-			break;
+		unsigned char id3Header[MP3_ID3V2_HEADER_LENGTH];
+		ret = fread(id3Header, sizeof(unsigned char), MP3_ID3V2_HEADER_LENGTH, fp);
```
Comment (vibhor-m) [inline framework/src/media/utils/MediaUtils.cpp:1039]: lets perform this fread operation also inside checkId3V2Header header. we have passed isFile (true/false) param.

File: framework/src/media/utils/MediaUtils.cpp [hunk 2]
```diff
@@ -617,18 +619,42 @@ static bool isMpeg2Ts(const unsigned char *buffer, size_t size)
 	return true;
 }
 
-static bool isMp3(const unsigned char *buffer, size_t size)
+static bool checkId3V2Header(const unsigned char *buffer, size_t size, bool isFile, FILE *fp = nullptr)
```
Comment (vibhor-m) [inline framework/src/media/utils/MediaUtils.cpp:622]: only fp pointer is enough, if fp is NULL lets use buffer else consider file

Comment (aadotverma) [reply framework/src/media/utils/MediaUtils.cpp]: Changes done

**Review by vibhor-m — APPROVED** [review]
code changes are OK

**Review by jsdosa — APPROVED** [review]
LGTM.

**Pull request merged**